joblibspark/backend.py

```diff
@@ -92,6 +94,36 @@ def __init__(self, **backend_args):
         except ImportError:
             self._ipython = None
 
+        self._support_stage_scheduling = self._is_support_stage_scheduling()
+        self._resource_profile = self._create_resource_profile(num_cpus_worker_node,
+                                                               num_gpus_worker_node)
+
+    def _is_support_stage_scheduling(self):
+        spark_master = self._spark_context.master
+        is_spark_local_mode = spark_master == "local" or spark_master.startswith("local[")
+        if is_spark_local_mode:
+            support_stage_scheduling = False
+            warnings.warn("Spark local mode doesn't support stage-level scheduling.")
+        else:
+            support_stage_scheduling = hasattr(
+                self._spark_context.parallelize([1]), "withResources"
+            )
+            warnings.warn("Spark version does not support stage-level scheduling.")
+        return support_stage_scheduling
+
+    def _create_resource_profile(self,
+                                 num_cpus_worker_node,
+                                 num_gpus_worker_node) -> Optional[object]:
+        resource_profile = None
+        if num_cpus_worker_node is not None and num_gpus_worker_node is not None:
```

Comment:
Let's do fallback like this:

```suggestion
        default_cpus_per_task = int(spark_conf.get("spark.task.cpus", "1"))
        default_gpus_per_task = int(spark_conf.get("spark.task.resource.gpu.amount", "0"))
        num_cpus_per_spark_task = num_cpus_per_spark_task or default_cpus_per_task
        num_gpus_per_spark_task = num_gpus_per_spark_task or default_gpus_per_task
```

Reply:
and let's make test on databricks GPU cluster to verify it works.